common/flatpak-installation.c

```diff
@@ -1227,8 +1229,12 @@ list_remotes_for_configured_remote (FlatpakInstallation *self,
       !types_filter[FLATPAK_REMOTE_TYPE_LAN])
     return TRUE;
 
+  repo = flatpak_dir_get_repo (dir);
+  if (repo == NULL)
+    return TRUE;
```

Comment:
I am still trying to wrap my head around, in what cases a FlatpakDir's `repo` member can be `NULL` ?

But looking at how `list_remotes_for_configued_remote` is called, I think this `repo==NULL` and return check can possibly live in the `flatpak_installation_list_remotes_by_type`. (there is already a `repo != NULL` check there).

Reply:
Tested the crash with https://paste.fedoraproject.org/paste/Ta~4EylNh0AJUyo9WcbCpg

Seems to fix it.

Reply:
> I am still trying to wrap my head around, in what cases a FlatpakDir's `repo` member can be `NULL` ?

I’m not 100% sure about the semantics, but I think it’s basically when a `FlatpakDir` is created with a repository which doesn’t exist. For example, the first time when flatpak is started. Or if a user `FlatpakDir` is created for a user who doesn’t have `~/.local/share/flatpak`.

> But looking at how `list_remotes_for_configued_remote` is called, I think this `repo==NULL` and return check can possibly live in the `flatpak_installation_list_remotes_by_type`. (there is already a `repo != NULL` check there).

It could, but I think it’s better to keep the check nearer to the use of the `dir`, i.e. where I’ve put it in `list_remotes_for_configured_remote()`. Then we don’t get surprises if that function is called from somewhere else (and we forget to add a check there) in future.